packages/core/src/application_ref.ts

```diff
@@ -177,11 +180,15 @@ export function assertPlatform(requiredToken: any): PlatformRef {
   const platform = getPlatform();
 
   if (!platform) {
-    throw new Error('No platform exists!');
+    const errorMessage =
+        (typeof ngDevMode === 'undefined' || ngDevMode) ? 'No platform exists!' : '';
+    throw new RuntimeError(RuntimeErrorCode.PLATFORM_NOT_FOUND, errorMessage);
   }
 
-  if (!platform.injector.get(requiredToken, null)) {
-    throw new Error(
+  if (!platform.injector.get(requiredToken, null) &&
+      (typeof ngDevMode === 'undefined' || ngDevMode)) {
```

Comment:
@ramthir tests in Google's codebase indicated a minor problem with this change where optimizer logic treats the `!platform.injector.get(requiredToken, null)` as always `true`. Could you please update the code to swap the operands:
```suggestion
  if ((typeof ngDevMode === 'undefined' || ngDevMode) && 
        !platform.injector.get(requiredToken, null)) {
```